src/web/app/user-join-page.component.spec.ts

```diff
@@ -142,27 +153,28 @@ describe('UserJoinPageComponent', () => {
   });
 
   it('should stop loading if user is not logged in', () => {
-    spyOn(courseService, 'getJoinCourseStatus').and.returnValue(of({
-      hasJoined: true,
-      userId: '',
-    }));
-
-    component.ngOnInit();
-
-    expect(component.isLoading).toBeFalsy();
-  });
+    spyOn(authService, 'getAuthUser').and.returnValue(
+      of({
+        user: {
```

Comment:
This is not a correct test data. The test case says "user is not logged in", so there cannot be any `user` value here.